src/gurt/dlog.c

```diff
@@ -387,6 +387,77 @@ log_exceed_threshold(void)
 	return mst.log_size + mst.log_buf_nob >= mst.log_size_max;
 }
 
+static int
+log_rotate(void)
+{
+	int rc = 0;
+
+	/* exceeds the size threshold, rename the current log file
+	 * as backup, create a new log file.
+	 */
+	if (!mst.log_old) {
+		rc = asprintf(&mst.log_old, "%s.old", mst.log_file);
+		if (rc < 0) {
+			dlog_print_err(errno, "failed to alloc name\n");
+			return -1;
+		}
+	}
+
+	if (mst.log_old_fd >= 0) {
+		close(mst.log_old_fd);
+		mst.log_old_fd = -1;
+	}
+
+	/* remove the backup log file */
+	rc = unlink(mst.log_old);
+	if (rc && errno != ENOENT) {
+		dlog_print_err(errno, "failed to unlink old file\n");
+		return -1;
+	}
```

Comment:
You don't need to do this, the rename will clobber any existing file.

Reply:
Removed